src/spritecache.cpp

```diff
@@ -72,6 +71,37 @@ static SpriteCache *AllocateSpriteCache(uint index)
 	return GetSpriteCache(index);
 }
 
+/**
+ * Get the cached SpriteFile given the name of the file.
+ * @param filename The name of the file at the disk.
+ * @return The SpriteFile or \c null.
+ */
+static SpriteFile *GetCachedSpriteFileByName(const std::string &filename) {
+	for (auto &f : _sprite_files) {
+		if (f->GetFilename() == filename) {
+			return f.get();
+		}
+	}
+	return nullptr;
+}
+
+/**
+ * Open/get the SpriteFile that is cached for use in the sprite cache.
+ * @param filename      Name of the file at the disk.
+ * @param subdir        The sub directory to search this file in.
+ * @param palette_remap Whether a palette remap needs to be performed for this file.
+ * @return The reference to the SpriteCache.
+ */
+SpriteFile &OpenCachedSpriteFile(const std::string &filename, Subdirectory subdir, bool palette_remap)
```

Comment:
This is (almost) the only function that doesn't return a pointer. Any particular reason why?

Reply:
My reasoning has been: if it can be `nullptr`, then make it a pointer. If it cannot be `nullptr` make it reference. Though that is my C++03? brain talking; I have no real clue about the nuances of all the new pointer types.